arch/lkl/lib/string.c

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#include <linux/string.h>
+#include <linux/export.h>
+
+#undef memcpy
+#undef memset
+#undef memmove
+
+__visible void *memcpy(void *dest, const void *src, size_t count)
+{
+	return __memcpy(dest, src, count);
+}
+EXPORT_SYMBOL(memcpy);
```

Comment:
Shouldn't this redirect already happen via the preprocessor using the `#defines` at https://github.com/lkl/linux/blob/master/arch/lkl/include/asm/string.h#L68 ? Are you building with or without `CONFIG_KASAN`?

Reply:
> Shouldn't this redirect already happen via the preprocessor using the #defines at https://github.com/lkl/linux/blob/master/arch/lkl/include/asm/string.h#L68 ?

There is code that uses `__builtin_memcpy` directly. Our `#define memcpy` will have no effect on such usage. The compiler will generate direct calls to `memcpy`, which is an undefined symbol because we never define it.

https://github.com/lkl/linux/blob/11f5b7eae2c2058cdb3710351117ac3e1c68764c/lib/string.c#L113

> Are you building with or without CONFIG_KASAN?

I built without CONFIG_KASAN and found these undefined symbols.

My patch was tested with and without CONFIG_KASAN. It should work either way.

Reply:
Thanks for the explanation, I wasn't familiar with how the compiler handled `__builtin_memcpy`. This change looks reasonable to me, although can't we now drop those `#define`s?